### ISSUE TITLE
[Accessibility]: Screen reader doesn't read disabled action item labels

### ISSUE THREAD

**Issue by tjkandala** [sourcegraph/sourcegraph#36852]
### Audit type

Screen reader navigation

### User journey audit issue

#34056

### Problem description

When focus is on a disabled action item, VoiceOver doesn't read the `aria-label`.

### Expected behavior

VoiceOver should read the `aria-label` in both enabled and disabled states.

### Additional details

![Screen Shot 2022-06-06 at 10 31 07 AM](https://user-images.githubusercontent.com/37420160/172705625-956e1e93-2aa1-42ac-82a5-bf7094216c7d.png)

### Assigning labels

- [X] Please give this issue an estimate by applying a label like `estimate/Xd`, where X is the estimated number of days it will take to complete.
- [X] If this issue is specific to a specific Sourcegraph product, please assign the appropriate team label to this issue.
- [X] If this issue will require input from designers in order to complete, please assign the label `needs-design`.
- [X] If you are confident that this issue should be fixed by GitStart, please assign the label `gitstart`.

### Owner

This issue will be fixed by my team, I have assigned a relevant member to it, or I will do so in the near future.

**Comment by sourcegraph-bot-2** [comment]
Heads up @muratsu @jjinnii @ryankscott - the "team/integrations" label was applied to this issue.
